tests/RemoteVerifySpec.hs

```diff
@@ -88,8 +91,22 @@ testSingleModule = do
 
 testUnsortedModules :: Spec
 testUnsortedModules = do
-  eReplState0 <- runIO $ loadCode
-    [text|
+  replState0 <- runIO $ either (error . show) id <$> loadCode code
+
+  it "loads when topologically sorted locally" $ do
+    stateModuleData "mod2" replState0 >>= (`shouldSatisfy` isRight)
+
+  resp <- runIO . runExceptT $ do
+    ModuleData mod1 _refs <- ExceptT $ stateModuleData "mod1" replState0
+    ModuleData mod2 _refs <- ExceptT $ stateModuleData "mod2" replState0
+    ExceptT . fmap (first show) . serveAndRequest 3001 $
+      Remote.Request [derefDef <$> mod2, derefDef <$> mod1] "mod2"
```

Comment:
An example of how to avoid partial pattern matches, as well as wanton `error` calls.